### PR TITLE
Dynalab 2.0

### PR DESCRIPTION
**Pull request by Ciroye**

### CODE REVIEW

File: frontends/web/src/containers/SubmitModel.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            </div>
+            <div className="mt-5">
+              <ul>
+                <li>
+                  <strong>Download dynalab 2.0</strong>
+                  <p>
+                    Please download{" "}
+                    <a
+                      download
+                      href="https://models-dynalab.s3.eu-west-3.amazonaws.com/MT/dynalab-base-mt.zip"
```
Comment (douwekiela) [inline frontends/web/src/containers/SubmitModel.js:81]: What does MT stand for here?

**Review by douwekiela — CHANGES_REQUESTED** [review]
I am missing the builder/evaluator code? I'm assuming that will have to change as well?

File: api/controllers/users.py [hunk 2]
```diff
@@ -465,3 +467,44 @@ def upload_user_profile_picture(credentials, id):
...
+
+    session = boto3.Session(
+        aws_access_key_id=config_file["aws_access_key_id"],
+        aws_secret_access_key=config_file["aws_secret_access_key"],
+        region_name=config_file["aws_region"],
+    )
+    s3_service = session.client("s3")
+    sqs_service = session.client("sqs")
+
+    if file_type != "application/zip":
```
Comment (douwekiela) [inline api/controllers/users.py:490]: Does this need more checks, eg with a reasonable size limit etc?

File: frontends/web/src/containers/SubmitModel.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                      this
+                    </a>{" "}
+                    project, and unzip in your own computer
+                  </p>
+                </li>
+                <li>
+                  <strong>Understand the folders</strong>
+                  <p>
+                    Inside the repository you will find a folder called app.
+                    Inside this folder you will find a folder called domain.
```
Comment (douwekiela) [inline frontends/web/src/containers/SubmitModel.js:93]: It's unclear from these instructions what "app" and "app/domain" are for. It's also unclear if model.py lives in app/domain/ or somewhere else

File: frontends/web/src/containers/SubmitModel.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                    only file that must be modified.
+                  </p>
+                </li>
+                <li>
+                  <strong>Single evaluation function</strong>
+                  <p>
+                    In this file you will find a class called "ModelController"
+                    with only two methods. The class contains only one mandatory
+                    method called "single_evaluation". As its name indicates
+                    this method must receive a single input and return a
```
Comment (douwekiela) [inline frontends/web/src/containers/SubmitModel.js:104]: Can we say more about what the input is? Basically an "example", right? I'd say "must receive a single example as input" then

File: frontends/web/src/containers/SubmitModel.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+                  <p>
+                    It is important to mention that you can create as many
+                    functions, classes, or variables as you consider necessary,
+                    just keeping in mind that the final result will be contained
+                    in the "single_evaluation" function of the "ModelController"
+                    class.
+                  </p>
+                </li>
+                <li>
+                  <strong>Where put your model and dependencies</strong>
```
Comment (douwekiela) [inline frontends/web/src/containers/SubmitModel.js:119]: Where **to**

File: frontends/web/src/containers/SubmitModel.js [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+                  <p>
+                    To store the different components of the model (weights,
+                    tokenizers, processors, etc) there is a folder called
+                    resources. Remember to put all the libraries necessary for
+                    your model in requirements.txt
```
Comment (douwekiela) [inline frontends/web/src/containers/SubmitModel.js:124]: missing final period

File: frontends/web/src/containers/SubmitModel.js [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+                    )}
+                  </div>
+                </li>
+              </ul>
+            </div>
+            <div className="text-center">
+              <h2 className="home-cardgroup-header d-block font-weight-light mb-4 text-uppercase text-reset">
+                Have a question?
+              </h2>
+              <a href="mailto:dynabench@fb.com">
```
Comment (douwekiela) [inline frontends/web/src/containers/SubmitModel.js:177]: Hrm I think we should change this email (and others?) to mlcommons.org?

File: api/controllers/users.py (hunk 2 above)
Comment (Ciroye) [reply api/controllers/users.py:490]: good point, let me add a size limit checker

File: frontends/web/src/containers/SubmitModel.js (hunk 7 above)
Comment (Ciroye) [reply frontends/web/src/containers/SubmitModel.js:196]: let me create another PR only related with this

File: api/controllers/tasks.py [hunk 8]
```diff
@@ -560,6 +560,20 @@ def get_task(task_id_or_code):
     return util.json_encode(task)
 
 
+@bottle.get("/tasks/<task_id_or_code>/dynalab")
+def get_dynalab_task(task_id_or_code):
+    if task_id_or_code == "18":
```
[the remark this answers is not in the document]
Comment (douwekiela) [reply api/controllers/tasks.py:565]: (the reason is, if I have a local install I might still have a 'flores2' task but at a different id)

File: dynalab/app/app.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+
+app = FastAPI()
+
+app.include_router(model.router, prefix="/model", tags=["model"])
+
+
+@app.get("/")
+def read_root():
+    return {"Hello": "Welcome to the jungle"}
```
Comment (douwekiela) [inline dynalab/app/app.py:17]: This looks a bit unprofessional to me ;) Maybe just status: ok? Or do we even need this?

**Review by douwekiela — APPROVED** [review]
LGTM let's test it out

File: dynalab/app/domain/model.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+class ModelController:
+    def __init__(self) -> None:
+        self.initialized = True
```
Comment (gwenzek) [inline dynalab/app/domain/model.py:8]: could we have a more realistic example ? Maybe a sample for the NLI task ?

e.g. I want to know how to load model and other resources.

Comment (Ciroye) [reply dynalab/app/domain/model.py:8]: excellent idea, let's use the NLI one and create a readme-tutorial this week

File: api/controllers/users.py [hunk 11]
```diff
@@ -532,6 +532,7 @@ def model_upload_s3_dynalab_2(credentials):
                 "user_id": user_id,
                 "user_name": user_name,
                 "task_code": task_code,
+                "model_id": model_id,
```
Comment (gwenzek) [inline api/controllers/users.py:535]: thanks that's good progress. 
Could we also have the "model_secret" ? This is required to call `api.dynabench.org/models/{model_id}/download`

https://github.com/mlcommons/dynabench/blob/0d0b9f68ea4cc56e0b8799b60e2db4a66bd910fd/builder/utils/helpers.py#L67-L67

Comment (Ciroye) [reply api/controllers/users.py]: Done

File: api/controllers/users.py [hunk 12]
```diff
@@ -465,3 +467,44 @@ def upload_user_profile_picture(credentials, id):
...
+
+    if file_type != "application/zip":
+        bottle.abort(403, "It's not a zip file")
+    response = s3_service.put_object(
+        Body=upload.file,
+        Bucket=config_file["aws_s3_bucket_name"],
+        Key=f"models/{task_code}/{user_id}-{file_name}",
+        ContentType=upload.content_type,
+    )
+    queue = sqs_service.get_queue_url(QueueName=config_file["new_builder_sqs_queue"])
```
[the remark this answers is not in the document]
Comment (Ciroye) [reply api/controllers/users.py]: Done

**Pull request merged**